test/Extension/UrlExtensionTest.php

```diff
@@ -43,94 +43,15 @@ public function testRegistersUrlFunctionWithEngine()
     {
         $engine = $this->prophesize(Engine::class);
         $engine
-            ->registerFunction('url', [$this->extension, 'generateUrl'])
+            ->registerFunction('url', $this->urlHelper)
```

Comment:
I think we did it this way originally as the original versions of Plates we targeted didn't understand functors (i.e., you couldn't pass just an instance, but also needed to pass the function to invoke). Additionally, the rationale for the `generate()` method was because the notation when composing a functor as a property of another class makes invoking it awkward (e.g., `($this->urlHelper)(...)` vs `$this->urlHelper->generate(...)`). But I absolutely agree that, since this works, this is a good change to make here.

Reply:
If I remember well Plates also performs further tests when it encounters a callable array expressions...

Reply:
if a callable array is still preferred, I would use [$this->urlHelper, '__invoke'], it's less "elegant" but still skips one internal proxy call...

Reply:
No, I like your solution here. Fewer layers of indirection. I was just noting why it was the way it was originally, in case folks come here wondering why it changed.

Reply:
@weierophinney Can I just change the target develop branch just with the "edit" button here on gh web interface? I remember doing it once and having issues....thanks!

Reply:
@pine3ree Yes, you can change the target on github but you also need rebase your changes and push force.

Reply:
@webimpress rebased to develop and targeted to develop

Reply:
@webimpress there are 3 or 4 null-coalescing replacement to be made and maybe we could add a comment to remind the removal of the redundant test cases in v. 3.0

Reply:
null-coalescing: it was just a reminder-comment for myself, since it's unrelated it's better to use a different PR for those minor changes. :-) kind regards

Reply:
yeah, can submit separate PR to master - as improvement, no functionality change.